CMakeLists.txt

```diff
@@ -462,6 +462,10 @@ if(BUILD_STATIC_LIB)
 else()
   message(STATUS "Building libbinaryen as shared library.")
   add_library(binaryen SHARED)
+  if(LINUX)
+    # Disable interposition and resolve Binaryen symbols locally.
+    add_link_flag("-Bsymbolic")
```

Comment:
This part is not actually necessary for the workaround. It's just a small optimization for DSO-local calls.